common/src/main/java/org/apache/rocketmq/common/MixAll.java

```diff
@@ -407,12 +408,50 @@ private static String localhost() {
             InetAddress addr = InetAddress.getLocalHost();
             return addr.getHostAddress();
         } catch (Throwable e) {
+            try {
+                String candidatesHost = getLocalhostByNetworkInterface();
+                if (candidatesHost != null)
+                    return candidatesHost;
+
+            } catch (Exception ignored) {
+            }
+
             throw new RuntimeException("InetAddress java.net.InetAddress.getLocalHost() throws UnknownHostException"
                 + FAQUrl.suggestTodo(FAQUrl.UNKNOWN_HOST_EXCEPTION),
                 e);
         }
     }
 
+    private static String getLocalhostByNetworkInterface() throws SocketException {
```

Comment:
RemotingUtil has already had a similar method, aka, getLocalAddress, is there any rational reason to create a duplicated one?

Reply:
I have marked it using FIXME label. reverse logic comparing to RemotingUtil method, we could consider to refactor it in RocketMQ 5.0.